### PR TITLE
Add IoT Connector functionality and config_io updates from ExoSense

### PR DESCRIPTION
**Pull request by alecjohanson**
This adds:
The "IOT_CONNECTOR" mode that allows for working as a Product-as-a-service. This will allow for quicker provisioning of devices.

A config_io converter, allowing users to configure the data channels from within ExoSense (instead of creating the config_io on their own)

### CODE REVIEW

**Comment by alecjohanson** [comment]
@betzrhodes

**Comment by betzrhodes** [comment]
@alecjohanson Is this ready for a new release? If so we should bump the version number. If it has no breaking changes, then v1.1.0?

**Comment by alecjohanson** [comment]
> @alecjohanson Is this ready for a new release? If so we should bump the version number. If it has no breaking changes, then v1.1.0?

Yeah, this is backwards compatible. Updated to 1.1.0

**Comment by betzrhodes** [comment]
Great will take a look and get update published soon.

**Comment by betzrhodes** [comment]
Example code seems to log errors with new library version: 

```
2019-09-30T21:11:40.671 +00:00 | [Status] | Device connected
2019-09-30T21:11:40.498 +00:00 | [Status] | Agent restarted: reload.
2019-09-30T21:11:40.507 +00:00 | [Agent] | fetching config_io
2019-09-30T21:11:40.507 +00:00 | [Agent] | headers: { "Content-Type": "application/x-www-form-urlencoded; charset=utf-8", "Accept": "application/x-www-form-urlencoded; charset=utf-8", "X-Exosite-CIK": "ZPSTyTUzYpa1v3n0sc6bbpCc6VvoI7mBQkgpMML9" }
2019-09-30T21:11:40.531 +00:00 | [Agent] | Server Response:
2019-09-30T21:11:40.531 +00:00 | [Agent] | 200
2019-09-30T21:11:40.531 +00:00 | [Agent] | config_io=hello+world
2019-09-30T21:11:40.531 +00:00 | [Agent] | writeConfigIO: hello world
2019-09-30T21:11:40.531 +00:00 | [Agent] | headers: { "Content-Type": "application/x-www-form-urlencoded; charset=utf-8", "Accept": "application/x-www-form-urlencoded; charset=utf-8", "X-Exosite-CIK": "ZPSTyTUzYpa1v3n0sc6bbpCc6VvoI7mBQkgpMML9" }
2019-09-30T21:11:40.531 +00:00 | [Agent] | Actual Input: hello world
2019-09-30T21:11:40.531 +00:00 | [Agent] | ERROR: Invalid JSON
2019-09-30T21:11:40.531 +00:00 | [Agent] | ERROR:   in _create_channel_converter_tabl agent_code:266
2019-09-30T21:11:40.531 +00:00 | [Agent] | ERROR:   from writeConfigIO agent_code:158
2019-09-30T21:11:40.531 +00:00 | [Agent] | ERROR:   from _pollConfigIOCallback agent_code:349
2019-09-30T21:11:42.199 +00:00 | [Agent] | writeData: { "press": 1014.0974, "humid": 39.311214, "temp": 26.005714 }
2019-09-30T21:11:42.199 +00:00 | [Agent] | headers: { "Content-Type": "application/x-www-form-urlencoded; charset=utf-8", "Accept": "application/x-www-form-urlencoded; charset=utf-8" }
2019-09-30T21:11:42.231 +00:00 | [Agent] | Server Response:
2019-09-30T21:11:42.231 +00:00 | [Agent] | 204
2019-09-30T21:11:42.231 +00:00 | [Agent]
```

**Comment by betzrhodes** [comment]
Need to update README include instructions to the new library version number - 
line 5 of readme

File: Exosite.agent.lib.nut [hunk 1]
```diff
@@ -24,10 +24,12 @@
 
 enum EXOSITE_MODES {
     MURANO_PRODUCT = "EXOSITE_MODE_MURANO_PRODUCT"
+    IOT_CONNECTOR = "EXOSITE_MODE_IOT_CONNECTOR"
 }
 
 class Exosite {
-      static VERSION = "1.0.0";
+      static IOT_CONNECTOR_ID = "g2bmsijv2ku800000";
```
Comment (betzrhodes) [inline Exosite.agent.lib.nut:31]: Is this always the same for everyone or is this a default connector id? Just curious

**Review by betzrhodes — COMMENTED** [review]
Automated tests passing. Code update looks good to me.

File: Exosite.agent.lib.nut (hunk 1 above)
Comment (alecjohanson) [reply Exosite.agent.lib.nut:31]: In the IOT_CONNECTOR mode it is the same. In the Murano_Product mode, the user needs to define it.

**Pull request merged**